ContestModel/src/org/icpc/tools/contest/model/internal/Contest.java

```diff
@@ -1809,7 +1809,10 @@ public void removeHiddenTeams() {
 				remove.add(clar);
 				foundClar = true;
 			}
-			teamId = clar.getToTeamId();
+
+			// For now, assume that clarifications to hidden teams are only
+			// sent to hidden teams. (i.e. no complex groups or hidden+non-hidden teams)
```

Comment:
I think this should also become part of the spec, so it's easier for someone to implement the spec and not have this edge case.

Reply:
See also my remark above